pylint/checkers/stdlib.py

```diff
@@ -494,8 +504,19 @@ def visit_call(self, node):
                 if inferred is astroid.Uninferable:
                     continue
                 if inferred.root().name == OPEN_MODULE:
-                    if getattr(node.func, "name", None) in OPEN_FILES:
+                    if (
+                        isinstance(node.func, astroid.Name)
+                        and node.func.name in OPEN_FILES_MODE
+                    ):
                         self._check_open_mode(node)
+                    if (
+                        isinstance(node.func, astroid.Name)
+                        and node.func.name in OPEN_FILES_ENCODING
+                    ) or (
+                        isinstance(node.func, astroid.Attribute)
```

Comment:
```suggestion
                        or isinstance(node.func, astroid.Attribute)
```
The parentheses aren't really necessary as `and` binds tighter than `or`.